include/boost/geometry/algorithms/detail/is_valid/has_duplicates.hpp

```diff
@@ -62,10 +56,10 @@ struct has_duplicates
                 typename Strategy::cs_tag
             > equal;
 
-        const_iterator it = boost::const_begin(view);
-        const_iterator next = it;
-        ++next;
-        for (; next != boost::const_end(view); ++it, ++next)
+        auto it = boost::begin(view);
+        auto const end = boost::end(view);
+        auto next = it;
+        for (++next; next != end; ++it, ++next)
```

Comment:
why not 
```
        auto it = boost::begin(view);
        auto next = it;
        for (++next; next != boost::end(view); ++it, ++next)
```
?

Reply:
Because AFAIU otherwise the new end iterator has to be created in each iteration.

Reply:
OK, if there is a creation in `boost::end(view)` I agree.